actionview/lib/action_view/helpers/text_helper.rb

```diff
@@ -467,18 +467,25 @@ def cut_excerpt_part(part_position, part, separator, options)
           radius   = options.fetch(:radius, 100)
           omission = options.fetch(:omission, "...")
 
-          part = part.split(separator)
```

Comment:
This was very expensive when `separator` is `""`, we end up generating a giant array of characters for a large string.